### ISSUE TITLE
Is there a timeline for when LXC support will appear in zstack?

### ISSUE THREAD

**Issue by bmullan** [zstackorg/zstack#23]
I saw the note:

>> Note: LXC/Docker and SR-IOV are not supported in the current version(0.6).

Is there any target timeline for when LXC support will be implemented?

thanks
brian

**Comment by bmullan** [comment]
With the release of LXC 1.x  there are now several language bindings for the LXC API:
    python3 (in-tree, long term support in 1.0.x)
    lua (in tree, long term support in 1.0.x)
    Go
    ruby
    python2
    Haskell

**Comment by zxwing** [comment]
what's the usage of LXC? want to use it as VMs?

**Comment by bmullan** [comment]
Yes.   I'd like to use LXC containers instead of hw virtualized machines
that you would get if you used vmware, virtualbox, kvm etc.

In LXC (linuxcontainers.org) I can download & install many different linux
distro's in my Ubuntu 14.04 host.   LXC has templates for ubuntu, centos,
opensuse, debian and others.  The only requirement obviously is that the
containers all run the same kernel has the "host".

With LXC I have had as many as 100 containers running (nested) on my Ubuntu
14.04 desktop.   I can't get more than a dozen or so hw VMs to run on same
hardware so scalability is a big factor.   That and LXC works great also on
ARM servers too.

So I think something like ZStack as an IaaS for LXC would be a great
solution for not only myself but many others.

Thanks.
Brian Mullan

**Comment by connermo** [comment]
IMHO LXC & ZStack are running on different levels of virtualization. You can have a CoreOS image running on ZStack to make LXC work.

**Comment by bmullan** [comment]
Not sure I understand?  zstack in its [architecture](https://docs.google.com/presentation/d/1mslQwl373-pZKuzZEbIIPd1mHG2ythI3RoyCVKRSeS8/edit#slide=id.g62dc262c2_015) describes an IaaS to me?   [zstack's workloads use KVM vm's](https://docs.google.com/presentation/d/1mslQwl373-pZKuzZEbIIPd1mHG2ythI3RoyCVKRSeS8/edit#slide=id.g62dc262c2_1177).

The only thing I was suggesting is that adding [LXD/LXC - mark shuttleworth's OpenStack Conference preso](https://www.youtube.com/watch?v=Ht7c6DAGZIU) support into zstack would seem to enable server density increases to public/private IaaS providers/users of zstack while still fully supporting KVM.

**Comment by zxwing** [comment]
You're right. ZStack now is IaaS software based on KVM (and VMWare, LXC in future).
For CoreOS and Docker, we consider they as the technology for software distributing. We will integrate them to ZStack  as an upper layer.

**Comment by sequoiar** [comment]
+1

**Comment by BoulidamAbdellah** [comment]
I want to know if I can use ZStack as a cloud platform and integrate it with LXC/LXD.
